src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java

```diff
@@ -0,0 +1,60 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.RoleDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RoleDaoImplTest extends AbstractTest {
+    private static final long ADMIN_ROLE_ID = 1L;
+    private static final long USER_ROLE_ID = 2L;
+    private static final String NON_EXIST_ROLE = "CUSTOMER";
+    private Role adminRole;
+    private Role userRole;
+    private RoleDao roleDao;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[] {Role.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        adminRole = new Role(Role.RoleName.ADMIN);
+        userRole = new Role(Role.RoleName.USER);
+        roleDao = new RoleDaoImpl(getSessionFactory());
+    }
+
+    @Test
+    void save_Ok() {
```

Comment:
Please also check other classes for same save_Ok test case naming
```suggestion
    void save_validRoles_Ok() {
```